Doesn't build anymore for webassembly target

I'm using the time package as a transitive dependency. The latest build seems incompatible with webassembly. Building my project with "cargo web build" yields the following error message:

       Compiling time v0.2.7
    error[E0308]: mismatched types
       --> /home/hc/.cargo/registry/src/github.com-1ecc6299db9ec823/time-0.2.7/src/utc_offset.rs:294:53
        |
    294 | fn try_local_offset_at(datetime: OffsetDateTime) -> Option<UtcOffset> {
        |    -------------------                              ^^^^^^^^^^^^^^^^^ expected enum `std::option::Option`, found `()`
        |    |
        |    implicitly returns `()` as its body has no tail or `return` expression
        |
        = note:   expected enum `std::option::Option<utc_offset::UtcOffset>`
                found unit type `()`

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -10,6 +10,7 @@ readme = "README.md"
 license = "MIT OR Apache-2.0"
 # TODO(upstream) Add GitHub Actions badge once rust-lang/crates.io#1838 is merged.
 description = "Date and time library. Fully interoperable with the standard library. Mostly compatible with #![no_std]."
+build = "build.rs"
 
 [package.metadata.docs.rs]
 all-features = true
@@ -26,6 +27,7 @@ std = ["libc", "winapi"]
 __doc = []
 
 [dependencies]
+cfg-if = "0.1"
 rand = { version = "0.7", optional = true, default-features = false }
 rustversion = "1"
 serde = { version = "1", optional = true, default-features = false, features = ["derive"] }
@@ -41,4 +43,4 @@ libc = { version = "0.2", optional = true }
 winapi = { version = "0.3", features = ["minwinbase", "minwindef", "timezoneapi"], optional = true }
 
 [target.'cfg(target_arch = "wasm32")'.dependencies]
-stdweb = "*"
+stdweb = "0.4"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -19,7 +19,7 @@ all-features = true
 default = ["deprecated", "std"]
 deprecated = []
 panicking-api = []
-std = ["libc", "winapi"]
+std = ["libc", "winapi", "stdweb"]
 
 # Internal usage. This is used when building for docs.rs and time-rs.github.io.
 # This feature should never be used by external users. It will likely be
@@ -43,4 +43,4 @@ libc = { version = "0.2", optional = true }
 winapi = { version = "0.3", features = ["minwinbase", "minwindef", "timezoneapi"], optional = true }
 
 [target.'cfg(target_arch = "wasm32")'.dependencies]
-stdweb = "0.4"
+stdweb = { version = "0.4", default-features = false, optional = true }

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -39,3 +39,6 @@ libc = { version = "0.2", optional = true }
 
 [target.'cfg(windows)'.dependencies]
 winapi = { version = "0.3", features = ["minwinbase", "minwindef", "timezoneapi"], optional = true }
+
+[target.'cfg(target_arch = "wasm32")'.dependencies]
+stdweb = "*"

--- a/build.rs
+++ b/build.rs
@@ -1,7 +1,7 @@
 use std::env;
 
 fn main() {
-    if env::var("COMPILING_UNDER_CARGO_WEB").is_ok() {
-        println!("cargo:rustc-cfg=cargoweb")
+    if env::var("COMPILING_UNDER_CARGO_WEB") == Ok("1".into()) {
+        println!("cargo:rustc-cfg=cargo_web")
     }
 }

--- a/build.rs
+++ b/build.rs
@@ -1,0 +1,7 @@
+use std::env;
+
+fn main() {
+    if env::var("COMPILING_UNDER_CARGO_WEB").is_ok() {
+        println!("cargo:rustc-cfg=cargoweb")
+    }
+}

--- a/src/utc_offset.rs
+++ b/src/utc_offset.rs
@@ -404,7 +404,8 @@ fn try_local_offset_at(datetime: OffsetDateTime) -> Option<UtcOffset> {
             fn filetime_to_secs(filetime: &FILETIME) -> i64 {
                 /// FILETIME represents 100-nanosecond intervals
                 const FT_TO_SECS: i64 = 10_000_000;
-                ((filetime.dwHighDateTime as i64) << 32 | filetime.dwLowDateTime as i64) / FT_TO_SECS
+                ((filetime.dwHighDateTime as i64) << 32 | filetime.dwLowDateTime as i64) /
+                    FT_TO_SECS
             }
 
             /// Convert an `OffsetDateTime` to a `SYSTEMTIME`.

--- a/src/utc_offset.rs
+++ b/src/utc_offset.rs
@@ -8,7 +8,7 @@ use crate::{
 };
 use core::fmt::{self, Display};
 
-#[cfg(target_arch = "wasm32")]
+#[cfg(cargoweb)]
 use stdweb::js;
 
 /// An offset from UTC.
@@ -295,185 +295,177 @@ impl Display for UtcOffset {
 #[cfg(feature = "std")]
 #[allow(clippy::too_many_lines)] //
 fn try_local_offset_at(datetime: OffsetDateTime) -> Option<UtcOffset> {
-    #[cfg(not(target_arch = "wasm32"))]
+    #[cfg(not(cargoweb))]
     use core::{convert::TryInto, mem};
 
-    #[cfg(target_family = "unix")]
-    {
-        /// Convert the given Unix timestamp to a `libc::tm`. Returns `None` on
+    cfg_if::cfg_if! {
+    if #[cfg(target_family = "unix")] {
+            /// Convert the given Unix timestamp to a `libc::tm`. Returns `None` on
         /// any error.
-        fn timestamp_to_tm(timestamp: i64) -> Option<libc::tm> {
-            extern "C" {
-                fn tzset();
-            }
+            fn timestamp_to_tm(timestamp: i64) -> Option<libc::tm> {
+                extern "C" {
+                    fn tzset();
+                }
 
-            let timestamp = timestamp.try_into().ok()?;
+                let timestamp = timestamp.try_into().ok()?;
 
-            // Safety: Plain old data.
-            #[allow(unsafe_code)]
-            let mut tm = unsafe { mem::zeroed() };
+                // Safety: Plain old data.
+                #[allow(unsafe_code)]
+                    let mut tm = unsafe { mem::zeroed() };
 
-            // Update timezone information from system. `localtime_r` does not
-            // do this for us.
-            //
-            // Safety: tzset is thread-safe.
-            #[allow(unsafe_code)]
-            unsafe {
-                tzset();
-            }
+                // Update timezone information from system. `localtime_r` does not
+                // do this for us.
+                //
+                // Safety: tzset is thread-safe.
+                #[allow(unsafe_code)]
+                    unsafe {
+                    tzset();
+                }
 
-            // Safety: We are calling a system API, which mutates the `tm`
-            // variable. If a null pointer is returned, an error occurred.
-            #[allow(unsafe_code)]
-            let tm_ptr = unsafe { libc::localtime_r(&timestamp, &mut tm) };
+                // Safety: We are calling a system API, which mutates the `tm`
+                // variable. If a null pointer is returned, an error occurred.
+                #[allow(unsafe_code)]
+                    let tm_ptr = unsafe { libc::localtime_r(&timestamp, &mut tm) };
 
-            if tm_ptr.is_null() {
-                None
-            } else {
-                Some(tm)
-            }
-        }
-
-        let tm = timestamp_to_tm(datetime.timestamp())?;
-
-        // `tm_gmtoff` extension
-        #[cfg(not(target_os = "solaris"))]
-        {
-            return tm.tm_gmtoff.try_into().ok().map(UtcOffset::seconds);
-        }
-
-        // No `tm_gmtoff` extension
-        #[cfg(target_os = "solaris")]
-        {
-            use crate::Date;
-            use core::convert::TryFrom;
-
-            let mut tm = tm;
-            if tm.tm_sec == 60 {
-                // Leap seconds are not currently supported.
-                tm.tm_sec = 59;
-            }
-
-            let local_timestamp =
-                Date::try_from_yo(1900 + tm.tm_year, u16::try_from(tm.tm_yday).ok()? + 1)
-                    .ok()?
-                    .try_with_hms(
-                        tm.tm_hour.try_into().ok()?,
-                        tm.tm_min.try_into().ok()?,
-                        tm.tm_sec.try_into().ok()?,
-                    )
-                    .ok()?
-                    .assume_utc()
-                    .timestamp();
-
-            return (local_timestamp - datetime.timestamp())
-                .try_into()
-                .ok()
-                .map(UtcOffset::seconds);
-        }
-    }
-
-    #[cfg(target_family = "windows")]
-    {
-        use crate::offset;
-        use winapi::{
-            shared::minwindef::FILETIME,
-            um::{
-                minwinbase::SYSTEMTIME,
-                timezoneapi::{SystemTimeToFileTime, SystemTimeToTzSpecificLocalTime},
-            },
-        };
-
-        /// Convert a `SYSTEMTIME` to a `FILETIME`. Returns `None` if any error
-        /// occurred.
-        fn systemtime_to_filetime(systime: &SYSTEMTIME) -> Option<FILETIME> {
-            // Safety: We only read `ft` if it is properly initialized.
-            #[allow(unsafe_code, deprecated)]
-            let mut ft = unsafe { mem::uninitialized() };
-
-            // Safety: `SystemTimeToFileTime` is thread-safe.
-            #[allow(unsafe_code)]
-            {
-                if 0 == unsafe { SystemTimeToFileTime(systime, &mut ft) } {
-                    // failed
+                if tm_ptr.is_null() {
                     None
                 } else {
-                    Some(ft)
+                    Some(tm)
                 }
             }
-        }
 
-        /// Convert a `FILETIME` to an `i64`, representing a number of seconds.
-        fn filetime_to_secs(filetime: &FILETIME) -> i64 {
-            /// FILETIME represents 100-nanosecond intervals
-            const FT_TO_SECS: i64 = 10_000_000;
-            ((filetime.dwHighDateTime as i64) << 32 | filetime.dwLowDateTime as i64) / FT_TO_SECS
-        }
+            let tm = timestamp_to_tm(datetime.timestamp())?;
 
-        /// Convert an `OffsetDateTime` to a `SYSTEMTIME`.
-        fn offset_to_systemtime(datetime: OffsetDateTime) -> SYSTEMTIME {
-            let (month, day_of_month) = datetime.to_offset(offset!(UTC)).month_day();
-            SYSTEMTIME {
-                wYear: datetime.year() as u16,
-                wMonth: month as u16,
-                wDay: day_of_month as u16,
-                wDayOfWeek: 0, // ignored
-                wHour: datetime.hour() as u16,
-                wMinute: datetime.minute() as u16,
-                wSecond: datetime.second() as u16,
-                wMilliseconds: datetime.millisecond(),
+            // `tm_gmtoff` extension
+            #[cfg(not(target_os = "solaris"))]
+                {
+                    tm.tm_gmtoff.try_into().ok().map(UtcOffset::seconds)
+                }
+
+            // No `tm_gmtoff` extension
+            #[cfg(target_os = "solaris")]
+                {
+                    use crate::Date;
+                    use core::convert::TryFrom;
+
+                    let mut tm = tm;
+                    if tm.tm_sec == 60 {
+                        // Leap seconds are not currently supported.
+                        tm.tm_sec = 59;
+                    }
+
+                    let local_timestamp =
+                        Date::try_from_yo(1900 + tm.tm_year, u16::try_from(tm.tm_yday).ok()? + 1)
+                            .ok()?
+                            .try_with_hms(
+                                tm.tm_hour.try_into().ok()?,
+                                tm.tm_min.try_into().ok()?,
+                                tm.tm_sec.try_into().ok()?,
+                            )
+                            .ok()?
+                            .assume_utc()
+                            .timestamp();
+
+                    (local_timestamp - datetime.timestamp())
+                        .try_into()
+                        .ok()
+                        .map(UtcOffset::seconds)
+                }
+        } else if #[cfg(target_family = "windows")] {
+            use crate::offset;
+            use winapi::{
+                shared::minwindef::FILETIME,
+                um::{
+                    minwinbase::SYSTEMTIME,
+                    timezoneapi::{SystemTimeToFileTime, SystemTimeToTzSpecificLocalTime},
+                },
+            };
+
+            /// Convert a `SYSTEMTIME` to a `FILETIME`. Returns `None` if any error
+        /// occurred.
+            fn systemtime_to_filetime(systime: &SYSTEMTIME) -> Option<FILETIME> {
+                // Safety: We only read `ft` if it is properly initialized.
+                #[allow(unsafe_code, deprecated)]
+                    let mut ft = unsafe { mem::uninitialized() };
+
+                // Safety: `SystemTimeToFileTime` is thread-safe.
+                #[allow(unsafe_code)]
+                    {
+                        if 0 == unsafe { SystemTimeToFileTime(systime, &mut ft) } {
+                            // failed
+                            None
+                        } else {
+                            Some(ft)
+                        }
+                    }
             }
-        }
 
-        // This function falls back to UTC if any system call fails.
-        let systime_utc = offset_to_systemtime(datetime.to_offset(offset!(UTC)));
+            /// Convert a `FILETIME` to an `i64`, representing a number of seconds.
+            fn filetime_to_secs(filetime: &FILETIME) -> i64 {
+                /// FILETIME represents 100-nanosecond intervals
+                const FT_TO_SECS: i64 = 10_000_000;
+                ((filetime.dwHighDateTime as i64) << 32 | filetime.dwLowDateTime as i64) / FT_TO_SECS
+            }
 
-        // Safety: `local_time` is only read if it is properly initialized, and
-        // `SystemTimeToTzSpecificLocalTime` is thread-safe.
-        #[allow(unsafe_code)]
-        let systime_local = unsafe {
-            #[allow(deprecated)]
-            let mut local_time = mem::uninitialized();
-            if 0 == SystemTimeToTzSpecificLocalTime(
-                core::ptr::null(), // use system's current timezone
-                &systime_utc,
-                &mut local_time,
-            ) {
-                // call failed
-                return None;
+            /// Convert an `OffsetDateTime` to a `SYSTEMTIME`.
+            fn offset_to_systemtime(datetime: OffsetDateTime) -> SYSTEMTIME {
+                let (month, day_of_month) = datetime.to_offset(offset!(UTC)).month_day();
+                SYSTEMTIME {
+                    wYear: datetime.year() as u16,
+                    wMonth: month as u16,
+                    wDay: day_of_month as u16,
+                    wDayOfWeek: 0, // ignored
+                    wHour: datetime.hour() as u16,
+                    wMinute: datetime.minute() as u16,
+                    wSecond: datetime.second() as u16,
+                    wMilliseconds: datetime.millisecond(),
+                }
+            }
+
+            // This function falls back to UTC if any system call fails.
+            let systime_utc = offset_to_systemtime(datetime.to_offset(offset!(UTC)));
+
+            // Safety: `local_time` is only read if it is properly initialized, and
+            // `SystemTimeToTzSpecificLocalTime` is thread-safe.
+            #[allow(unsafe_code)]
+                let systime_local = unsafe {
+                #[allow(deprecated)]
+                    let mut local_time = mem::uninitialized();
+                if 0 == SystemTimeToTzSpecificLocalTime(
+                    core::ptr::null(), // use system's current timezone
+                    &systime_utc,
+                    &mut local_time,
+                ) {
+                    // call failed
+                    return None;
+                } else {
+                    local_time
+                }
+            };
+
+            // Convert SYSTEMTIMEs to FILETIMEs so we can perform arithmetic on them.
+            let ft_system = systemtime_to_filetime(&systime_utc)?;
+            let ft_local = systemtime_to_filetime(&systime_local)?;
+
+            let diff_secs = filetime_to_secs(&ft_local) - filetime_to_secs(&ft_system);
+
+            diff_secs.try_into().ok().map(UtcOffset::seconds)
+        } else if #[cfg(cargoweb)] {
+            let timestamp_utc: i64 = datetime.timestamp();
+            let t1 = (timestamp_utc & 0xFFFFFFFF) as i32;
+            let t2 = (timestamp_utc >> 32) as i32;
+            let timezone_offset = js! {
+                return new Date(@{t2} << 32 + @{t1}).getTimezoneOffset() * -60;
+            };
+            if let Ok(seconds) = stdweb::unstable::TryInto::try_into(timezone_offset) {
+                Some(UtcOffset::seconds(seconds))
             } else {
-                local_time
+                None
             }
-        };
-
-        // Convert SYSTEMTIMEs to FILETIMEs so we can perform arithmetic on them.
-        let ft_system = systemtime_to_filetime(&systime_utc)?;
-        let ft_local = systemtime_to_filetime(&systime_local)?;
-
-        let diff_secs = filetime_to_secs(&ft_local) - filetime_to_secs(&ft_system);
-
-        return diff_secs.try_into().ok().map(UtcOffset::seconds);
-    }
-
-    #[cfg(target_arch = "wasm32")]
-    {
-        let timestamp_utc: i64 = datetime.timestamp();
-        let t1 = (timestamp_utc & 0xFFFFFFFF) as i32;
-        let t2 = (timestamp_utc >> 32) as i32;
-        let timezone_offset = js! {
-            var ts_utc = @{t2} << 32 + @{t1};
-            var d = new Date(ts_utc);
-            return d.getTimezoneOffset() * -60;
-        };
-        if let Ok(seconds) = stdweb::unstable::TryInto::try_into(timezone_offset) {
-            return Some(UtcOffset::seconds(seconds));
         } else {
-            return None;
+            None
         }
     }
-
-    #[allow(unreachable_code)]
-    None
 }
 
 #[cfg(test)]

--- a/src/utc_offset.rs
+++ b/src/utc_offset.rs
@@ -339,7 +339,7 @@ fn try_local_offset_at(datetime: OffsetDateTime) -> Option<UtcOffset> {
         // `tm_gmtoff` extension
         #[cfg(not(target_os = "solaris"))]
         {
-            tm.tm_gmtoff.try_into().ok().map(UtcOffset::seconds)
+            return tm.tm_gmtoff.try_into().ok().map(UtcOffset::seconds);
         }
 
         // No `tm_gmtoff` extension
@@ -366,10 +366,10 @@ fn try_local_offset_at(datetime: OffsetDateTime) -> Option<UtcOffset> {
                     .assume_utc()
                     .timestamp();
 
-            (local_timestamp - datetime.timestamp())
+            return (local_timestamp - datetime.timestamp())
                 .try_into()
                 .ok()
-                .map(UtcOffset::seconds)
+                .map(UtcOffset::seconds);
         }
     }
 
@@ -452,7 +452,7 @@ fn try_local_offset_at(datetime: OffsetDateTime) -> Option<UtcOffset> {
 
         let diff_secs = filetime_to_secs(&ft_local) - filetime_to_secs(&ft_system);
 
-        diff_secs.try_into().ok().map(UtcOffset::seconds)
+        return diff_secs.try_into().ok().map(UtcOffset::seconds);
     }
 
     #[cfg(target_arch = "wasm32")]
@@ -466,11 +466,14 @@ fn try_local_offset_at(datetime: OffsetDateTime) -> Option<UtcOffset> {
             return d.getTimezoneOffset() * -60;
         };
         if let Ok(seconds) = stdweb::unstable::TryInto::try_into(timezone_offset) {
-            Some(UtcOffset::seconds(seconds))
+            return Some(UtcOffset::seconds(seconds));
         } else {
-            None
+            return None;
         }
     }
+
+    #[allow(unreachable_code)]
+    None
 }
 
 #[cfg(test)]

--- a/src/utc_offset.rs
+++ b/src/utc_offset.rs
@@ -7,8 +7,7 @@ use crate::{
     DeferredFormat, Duration,
 };
 use core::fmt::{self, Display};
-
-#[cfg(cargoweb)]
+#[cfg(cargo_web)]
 use stdweb::js;
 
 /// An offset from UTC.
@@ -293,15 +292,14 @@ impl Display for UtcOffset {
 /// Attempt to obtain the system's UTC offset. If the offset cannot be
 /// determined, `None` is returned.
 #[cfg(feature = "std")]
-#[allow(clippy::too_many_lines)] //
+#[allow(clippy::too_many_lines)]
 fn try_local_offset_at(datetime: OffsetDateTime) -> Option<UtcOffset> {
-    #[cfg(not(cargoweb))]
-    use core::{convert::TryInto, mem};
-
     cfg_if::cfg_if! {
-    if #[cfg(target_family = "unix")] {
-            /// Convert the given Unix timestamp to a `libc::tm`. Returns `None` on
-        /// any error.
+        if #[cfg(target_family = "unix")] {
+            use core::{convert::TryInto, mem};
+
+            /// Convert the given Unix timestamp to a `libc::tm`. Returns `None`
+            /// on any error.
             fn timestamp_to_tm(timestamp: i64) -> Option<libc::tm> {
                 extern "C" {
                     fn tzset();
@@ -313,8 +311,8 @@ fn try_local_offset_at(datetime: OffsetDateTime) -> Option<UtcOffset> {
                 #[allow(unsafe_code)]
                     let mut tm = unsafe { mem::zeroed() };
 
-                // Update timezone information from system. `localtime_r` does not
-                // do this for us.
+                // Update timezone information from system. `localtime_r` does
+                // not do this for us.
                 //
                 // Safety: tzset is thread-safe.
                 #[allow(unsafe_code)]
@@ -372,6 +370,7 @@ fn try_local_offset_at(datetime: OffsetDateTime) -> Option<UtcOffset> {
                         .map(UtcOffset::seconds)
                 }
         } else if #[cfg(target_family = "windows")] {
+            use core::{convert::TryInto, mem};
             use crate::offset;
             use winapi::{
                 shared::minwindef::FILETIME,
@@ -381,8 +380,8 @@ fn try_local_offset_at(datetime: OffsetDateTime) -> Option<UtcOffset> {
                 },
             };
 
-            /// Convert a `SYSTEMTIME` to a `FILETIME`. Returns `None` if any error
-        /// occurred.
+            /// Convert a `SYSTEMTIME` to a `FILETIME`. Returns `None` if any
+            /// error occurred.
             fn systemtime_to_filetime(systime: &SYSTEMTIME) -> Option<FILETIME> {
                 // Safety: We only read `ft` if it is properly initialized.
                 #[allow(unsafe_code, deprecated)]
@@ -400,7 +399,8 @@ fn try_local_offset_at(datetime: OffsetDateTime) -> Option<UtcOffset> {
                     }
             }
 
-            /// Convert a `FILETIME` to an `i64`, representing a number of seconds.
+            /// Convert a `FILETIME` to an `i64`, representing a number of
+            /// seconds.
             fn filetime_to_secs(filetime: &FILETIME) -> i64 {
                 /// FILETIME represents 100-nanosecond intervals
                 const FT_TO_SECS: i64 = 10_000_000;
@@ -444,25 +444,26 @@ fn try_local_offset_at(datetime: OffsetDateTime) -> Option<UtcOffset> {
                 }
             };
 
-            // Convert SYSTEMTIMEs to FILETIMEs so we can perform arithmetic on them.
+            // Convert SYSTEMTIMEs to FILETIMEs so we can perform arithmetic on
+            // them.
             let ft_system = systemtime_to_filetime(&systime_utc)?;
             let ft_local = systemtime_to_filetime(&systime_local)?;
 
             let diff_secs = filetime_to_secs(&ft_local) - filetime_to_secs(&ft_system);
 
             diff_secs.try_into().ok().map(UtcOffset::seconds)
-        } else if #[cfg(cargoweb)] {
-            let timestamp_utc: i64 = datetime.timestamp();
-            let t1 = (timestamp_utc & 0xFFFFFFFF) as i32;
-            let t2 = (timestamp_utc >> 32) as i32;
+        } else if #[cfg(cargo_web)] {
+            use stdweb::unstable::TryInto;
+
+            let timestamp_utc = datetime.timestamp();
+            let low_bits = (timestamp_utc & 0xFF_FF_FF_FF) as i32;
+            let high_bits = (timestamp_utc >> 32) as i32;
+
             let timezone_offset = js! {
-                return new Date(@{t2} << 32 + @{t1}).getTimezoneOffset() * -60;
+                return new Date(@{high_bits} << 32 + @{low_bits}).getTimezoneOffset() * -60;
             };
-            if let Ok(seconds) = stdweb::unstable::TryInto::try_into(timezone_offset) {
-                Some(UtcOffset::seconds(seconds))
-            } else {
-                None
-            }
+
+            timezone_offset.try_into().ok().map(UtcOffset::seconds)
         } else {
             None
         }

--- a/src/utc_offset.rs
+++ b/src/utc_offset.rs
@@ -8,6 +8,9 @@ use crate::{
 };
 use core::fmt::{self, Display};
 
+#[cfg(target_arch = "wasm32")]
+use stdweb::js;
+
 /// An offset from UTC.
 ///
 /// Guaranteed to store values up to ±23:59:59. Any values outside this range
@@ -292,6 +295,7 @@ impl Display for UtcOffset {
 #[cfg(feature = "std")]
 #[allow(clippy::too_many_lines)] //
 fn try_local_offset_at(datetime: OffsetDateTime) -> Option<UtcOffset> {
+    #[cfg(not(target_arch = "wasm32"))]
     use core::{convert::TryInto, mem};
 
     #[cfg(target_family = "unix")]
@@ -449,6 +453,23 @@ fn try_local_offset_at(datetime: OffsetDateTime) -> Option<UtcOffset> {
         let diff_secs = filetime_to_secs(&ft_local) - filetime_to_secs(&ft_system);
 
         diff_secs.try_into().ok().map(UtcOffset::seconds)
+    }
+
+    #[cfg(target_arch = "wasm32")]
+    {
+        let timestamp_utc: i64 = datetime.timestamp();
+        let t1 = (timestamp_utc & 0xFFFFFFFF) as i32;
+        let t2 = (timestamp_utc >> 32) as i32;
+        let timezone_offset = js! {
+            var ts_utc = @{t2} << 32 + @{t1};
+            var d = new Date(ts_utc);
+            return d.getTimezoneOffset() * -60;
+        };
+        if let Ok(seconds) = stdweb::unstable::TryInto::try_into(timezone_offset) {
+            Some(UtcOffset::seconds(seconds))
+        } else {
+            None
+        }
     }
 }
 

